appservice/src/deploy/localGitDeploy.ts

```diff
@@ -8,14 +8,15 @@ import * as git from 'simple-git/promise';
 import * as vscode from 'vscode';
 import { callWithMaskHandling, IActionContext } from 'vscode-azureextensionui';
 import { ext } from '../extensionVariables';
+import { IAppSettingsClient } from '../IAppSettingsClient';
+import { IDeploymentsClient } from '../IDeploymentsClient';
 import { localize } from '../localize';
-import { SiteClient } from '../SiteClient';
 import { nonNullProp } from '../utils/nonNull';
 import { openUrl } from '../utils/openUrl';
 import { verifyNoRunFromPackageSetting } from '../verifyNoRunFromPackageSetting';
 import { waitForDeploymentToComplete } from './waitForDeploymentToComplete';
 
-export async function localGitDeploy(client: SiteClient, fsPath: string, context: IActionContext): Promise<void> {
+export async function localGitDeploy(client: IDeploymentsClient & IAppSettingsClient, fsPath: string, context: IActionContext, branch: string = 'master', commit: boolean = false): Promise<void> {
```

Comment:
I think it's worth passing an options parameter.  You can make it like this:
`options: localGitOptions`

```
type localGitOptions {
   fsPath: string;
   branch?: string;
   commit?: boolean;
}
```

The `localGitDeploy` is getting a little cluttered in the parameters.

Reply:
Good idea